distributed/client.py

```diff
@@ -2332,14 +2301,16 @@ def map(
         if allow_other_workers and workers is None:
             raise ValueError("Only use allow_other_workers= if using workers=")
 
-        dsk = _MapLayer(
+        expr = _MapExpr(
             func,
             iterables,
             key=key,
             pure=pure,
-            **kwargs,
+            # FIXME: this doesn't look right
```

Comment:
Do we still need to address this?

Reply:
that's actually alright. the global annotations will be attached in graph_to_future